src/sage/matroids/matroids_plot_helpers.py

```diff
@@ -172,10 +172,11 @@ def it(M, B1, nB1, lps):
             cc = interval*j
             pts[i[j-1]] = (cc*pt1[0]+(1-cc)*pt2[0], cc*pt1[1]+(1-cc)*pt2[1])
     trilines = [list(set(x)) for x in lines if len(x) >= 3]
-    curvedlines = [list(set(list(x)).difference(set(lps)))
-                   for x in M.flats(2) if set(list(x)) not in trilines if
-                   len(list(x)) >= 3]
-    nontripts = [i for i in nB1 if i not in pts.keys()]
+    set_lps = set(lps)
+    curvedlines = [list(sx.difference(set_lps))
+                   for x in M.flats(2) if (sx := set(x)) not in trilines
```

Comment:
is this condition ever true? `trilines` appears to be a list of lists

Reply:
This part of the code looked strange before my changes. I am not sure what is intended by the author.